### PR TITLE
Framegraph: fix reporting of mapped memory

### PR DESCRIPTION
**Pull request by hevrard**
This change sets the memoryMapped flag of framegraph buffers/images
only when the mapped memory is actually overlapping a range of memory
that backs the buffer/image.

Also, when updating the buffer/image lookup tables, make sure to also
scan the sparse memory of each resource.

Also, stop assuming that there is always a non-sparse memory bound,
this removes some possible NPE.

Bug: b/175394793, b/174505556
Test: manual

### CODE REVIEW

**Comment by hevrard** [comment]
A gentle ping on this one

**Review by johnplate — APPROVED** [review]
LGTM, but should be reviewed also by someone who is more familiar with this code part.

File: gapis/api/vulkan/framegraph.go [hunk 1]
```diff
@@ -82,25 +107,61 @@ func newFramegraphImage(state *State, img *ImageObjectʳ) *api.FramegraphImage {
...
+							if !(mem.MappedLocation().IsNullptr()) {
+								memRange := memory.Range{
+									Base: uint64(block.MemoryOffset()),
+									Size: uint64(block.Size()),
+								}
+								mappedRange := memory.Range{
+									Base: uint64(mem.MappedOffset()),
+									Size: uint64(mem.MappedSize()),
+								}
+								memoryMapped = memoryMapped || memRange.Overlaps(mappedRange)
```
Comment (AWoloszyn) [inline gapis/api/vulkan/framegraph.go:162]: You should be able to drop out of this nested loop if coherent&&mapped.

Comment (hevrard) [reply gapis/api/vulkan/framegraph.go:172]: Added early exits in loops.

**Pull request merged**